exec/java-exec/src/main/java/org/apache/drill/exec/store/log/LogRecordReader.java

```diff
@@ -0,0 +1,750 @@
+        return new DoubleDefn(name, patternIndex);
+      case DATE:
+        return new DateDefn(name, patternIndex, formatConfig.getDateFormat(patternIndex));
+      case TIMESTAMP:
+        return new TimeStampDefn(name, patternIndex, formatConfig.getDateFormat(patternIndex));
+      case TIME:
+        return new TimeDefn(name, patternIndex, formatConfig.getDateFormat(patternIndex));
+      default:
+        throw UserException
+            .validationError()
+            .message("Undefined column types")
+            .addContext("Position", patternIndex)
+            .addContext("Field name", name)
+            .addContext("Type", typeName)
+            .build(logger);
+    }
+  }
+
+  private void defineVectors() {
+    for (int i = 0; i < columns.length; i++) {
+      try {
+        columns[i].define(outputMutator);
+      } catch (SchemaChangeException e) {
+        throw UserException
+            .systemError(e)
+            .message("Vector creation failed")
+            .build(logger);
+      }
+    }
+  }
+
+  @Override
+  public int next() {
+    rowIndex = 0;
+    while (nextLine()) {
+    }
+    return rowIndex;
+  }
+
+  private boolean nextLine() {
+    String line;
+    try {
+      line = reader.readLine();
+    } catch (IOException e) {
+      throw UserException
+          .dataReadError(e)
+          .message("Error reading file:")
+          .addContext("File", fileWork.getPath())
+          .build(logger);
+    }
+
+    if (line == null) {
+      return false;
+    }
+    Matcher lineMatcher = pattern.matcher(line);
+    if (lineMatcher.matches()) {
+      loadVectors(lineMatcher);
+      return rowIndex < BATCH_SIZE;
+    }
+
+    errorCount++;
+    if (errorCount < formatConfig.getMaxErrors()) {
+      logger.warn("Unmatached line: {}", line);
+    } else if (errorCount > formatConfig.getMaxErrors()) {
+      throw UserException.parseError()
+          .message("Too many errors.  Max error threshold exceeded.")
+          .addContext("Line", line)
+          .addContext("Line number", rowIndex)
+          .build(logger);
+    }
+    //If the user asked for the unmatched columns display them
+    if (unmatchedRows) {
+      //If the user asked for the unmatched columns AND other columns
+      if (columns.length > 1) {
+        columns[unmatchedColumnIndex].load(rowIndex, line);
+        rowIndex++;
+        return rowIndex < BATCH_SIZE;
+      } else {
+        //If the user ONLY asked for the unmatched columns
+        columns[unmatchedColumnIndex].load(unmatchedRowIndex, line);
+        unmatchedRowIndex++;
+        rowIndex = unmatchedRowIndex;
+        return unmatchedRowIndex < BATCH_SIZE;
+      }
+    }
+
+    return true;
+  }
+
+  private void loadVectors(Matcher m) {
+    String value = null;
+    /*if( unmatchedRows && columns.length == 1 ){
+      return;
+    }*/
+
+    for (int i = 0; i < columns.length; i++) {
+      //Skip the unmatched rows column
+      if( columns[i].name.equals("_unmatched_rows")){
```

Comment:
This is pretty expensive: doing a string compare for each column for each row. We also do this below.

Maybe define another column type for a "Special" column. Then, have every type implement `isSpecial()`. Now, you can simply skip `isSpecial()` columns rather than doing string comparisons.